src/app/features/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.ts

```diff
@@ -25,19 +33,54 @@ export class HydrateNewVehicleRecordComponent {
 
   get vehicle$(): Observable<VehicleTechRecordModel | undefined> {
     return this.technicalRecordService.editableVehicleTechRecord$.pipe(
-      tap(techRecord => {
-        if (!techRecord) this.navigateBack();
+      tap(vehicle => {
+        if (!vehicle) this.navigateBack();
+        this.vehicleType = vehicle?.techRecord[0].vehicleType;
       })
     );
   }
 
-  navigateBack() {
-    this.globalErrorService.clearErrors();
-    this.router.navigate(['..'], { relativeTo: this.route });
+  get customSectionForms(): Array<CustomFormGroup> {
+    if (this.summary && this.vehicleType) {
+      const commonCustomSections = [this.summary.body.form, this.summary.dimensions.form, this.summary.tyres.form, this.summary.weights.form];
+
+      switch (this.vehicleType) {
+        case VehicleTypes.PSV:
+          return [...commonCustomSections, this.summary.psvBrakes!.form];
+        case VehicleTypes.HGV:
+          return commonCustomSections;
+        case VehicleTypes.TRL:
+          return [...commonCustomSections, this.summary.trlBrakes!.form];
```

Comment:
Instead of using `!` maybe use the `?` operator.

Reply:
I think it might be a good use case here as the line above verifies that it is a TRL and we expect it to contain `trlBrakes`.

Reply:
Can't, this causes a whole chain of problems where undefined needs sticking in a bunch of types. We shouldn't be making tech records if we cant get the brakes sections surely?

Reply:
Hmm I guess, if enforcing typing is a rat's nest at this point then maybe this is the only option. Could you change this switch statement to a map instead though?

Reply:
Switch statements are - computationally - really minimal and instantly readable. Have you got an argument for why a map would be better? Besides it being less basic and long?